### PR TITLE
[i18n] Install typesafe-i18n & translates one component

### PR DESCRIPTION
**Pull request by rajatvijay**
Relates #2927

**Technical details**

This PR does the following things: 

1. Installs typesafe-i18n lib and customize config as per our linting rules.
2. i18n ImportUploadPage component.
3. Temporarily loads the translation file from App.svelte
4. Fallbacks to en in ja lang

**Screenshots**

Screenshot 2023-07-27 at 7 43 40 AM

## Checklist
 
- [x] My pull request has a descriptive title (not a vague title like `Update
  index.md`).
- [x] My pull request targets the `develop` branch of the repository
- [x] My commit messages follow [best practices][best_practices].
- [x] My code follows the established code style of the repository.
- [ ] I added tests for the changes I made (if applicable).
- [ ] I added or updated documentation (if applicable).
- [x] I tried running the project locally and verified that there are no
  visible errors.

[best_practices]:https://gist.github.com/robertpainsi/b632364184e70900af4ab688decf6f53

## Developer Certificate of Origin

<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>

### CODE REVIEW

**Comment by seancolsen** [comment]
@rajatvijay can you fix the merge conflicts in `App.svelte`

As a side note, I wish you had chosen a more trivial piece of text to translate for this initial PR, like "Shortcuts" in the nav header or something similar that's very unlikely to intersect with the work of other team members. Instead, you coincidentally chose to modify an area of the app that I am [currently ripping apart](https://github.com/centerofci/mathesar/pull/3050/files#diff-11e401ee47e8a7d417d9024e87fb356ac6519b0283c010d529b99879b23cab97). That means that one of us will need to manually propagate these changes over in order to resolve these merge conflicts.

File: mathesar_ui/src/App.svelte
Comment (seancolsen) [inline mathesar_ui/src/App.svelte:32]: Why the IIFE here? Should the go into an `onMount` function instead? Why `return true`? Why `.then` instead of `await`?

**Review by seancolsen — COMMENTED** [review]
I've only skimmed the diff so far. I had some trouble running Mathesar with these changes. I kept getting this strange error:

image

What do you make of this, @rajatvijay? I restarted Docker with the following command, which I would expect to be a sure-fire way of getting all the dependencies sorted out.

```
sudo docker compose -f docker-compose.yml -f docker-compose.dev.yml up dev-service --force-recreate --build dev-s
ervice
```

I didn't spend much time troubleshooting though. Do you have any ideas?

File: mathesar_ui/src/App.svelte
Comment (rajatvijay) [reply mathesar_ui/src/App.svelte]: - Re-wrote this to use async/await
- Not doing this in `onMount` since do not want to wait to fetch the file until the first render. 
- Not returning `true` anymore
- Anyways, in [this](https://github.com/centerofci/mathesar/pull/3102) PR this whole piece of code is going away.

Comment (rajatvijay) [reply mathesar_ui/src/App.svelte]: Here: 8e8f6c5

**Comment by rajatvijay** [comment]
> I've only skimmed the diff so far. I had some trouble running Mathesar with these changes. I kept getting this strange error:
> 
> image
> 
> What do you make of this, @rajatvijay? I restarted Docker with the following command, which I would expect to be a sure-fire way of getting all the dependencies sorted out.
> 
> ```
> sudo docker compose -f docker-compose.yml -f docker-compose.dev.yml up dev-service --force-recreate --build dev-s
> ervice
> ```
> 
> I didn't spend much time troubleshooting though. Do you have any ideas?

@seancolsen not sure. Not facing this on my machine. I ran the same command again for force re-create. 
I also ran it a new codespace and did not face this error either. 
Screenshot 2023-07-28 at 2 54 33 PM

Maybe check in the `node_modules` if this file exists. This will pin point if this is a package/installation problem or bundler issue.

**Comment by seancolsen** [comment]
Regarding the import error, I did some further troubleshooting and identified that the package was not installed. I was able to fix it by opening a shell in the container and running `npm ci`. But this is very weird to me. I've never had to do anything like that before after we've installed a new front end package.

@pavish would you be up for looking into this issue? I'm wondering if there may be a docker-related change that affected this behavior. And I'm nervous about merging this PR without chasing this problem down more thoroughly because we could end up breaking everyone's development environment. @pavish Are you able to checkout this PR, restart Docker with `--force-recreate`, and then run Mathesar? Or do you see an error like in my screenshot above? If you see that error, then I wonder if you might have some insight into how to fix it. If you don't see the error, then do you have any ideas what I should try to continue troubleshooting on my end, just to ensure nobody else hits this error after we merge this PR?

**Review by seancolsen — APPROVED** [review]
This looks good to me. I'm deferring some aspects of the review to your subsequent i18n PRs. For example, I'd like to try running in prod mode once the dust settles with the loading of translation files.

I'm ready to merge this once we do a bit more digging into the npm error to make sure that we're not going to break the workflow of other devs by merging.

**Comment by pavish** [comment]
> @pavish would you be up for looking into this issue? I'm wondering if there may be a docker-related change that affected this behavior.

[Relevant matrix chat](https://matrix.to/#/!UnujZDUxGuMrYdvgTU:matrix.mathesar.org/$2yCV64geGRiOGhkzhnv7XXhKOyXHDDn16h7f8BCBv1M?via=matrix.mathesar.org&via=matrix.org)

**Pull request merged**
